swig/csharp/CMakeLists.txt

```diff
@@ -9,226 +15,624 @@ set(GDAL_CSHARP_INSTALL_DIR
-    VERBATIM
-    DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/${_CSHARP_WRAPPER}.cpp ${_depends} ${CMAKE_CURRENT_BINARY_DIR}/gdal.snk
-            ${CMAKE_CURRENT_SOURCE_DIR}/AssemblyInfo.cs)
 
   install(
     TARGETS ${_CSHARP_WRAPPER}
     COMPONENT csharp
     DESTINATION ${GDAL_CSHARP_INSTALL_DIR})
+endfunction (gdal_csharp_wrap)
+
+# function for xxx_csharp.dll build
+function (gdal_csharp_dll)
+
+  # Setup
+  set(_options)
+  set(_oneValueArgs TARGET TARGET_SUBDIR WRAP_DLL PACKAGE_NAME)
+  set(_multiValueArgs DEPENDS SYSTEM_DEPENDS)
+  cmake_parse_arguments(_CSHARP "${_options}" "${_oneValueArgs}" "${_multiValueArgs}" ${ARGN})
+  get_filename_component(_root ${_CSHARP_TARGET} NAME_WE)
+
+  file(TO_NATIVE_PATH "${CMAKE_CURRENT_BINARY_DIR}/${_CSHARP_TARGET_SUBDIR}/${_root}.csproj" _CSHARP_PROJ)
+  file(TO_NATIVE_PATH "${CMAKE_CURRENT_BINARY_DIR}/${_CSHARP_TARGET_SUBDIR}" _CSHARP_PROJ_PATH)
+  file(TO_NATIVE_PATH "${CMAKE_CURRENT_BINARY_DIR}/${_CSHARP_TARGET_SUBDIR}/${_CSHARP_TARGET}" _CSHARP_DLL_TARGET)
+  set(CSC_OPTIONS /unsafe /debug:full /target:library /out:${_CSHARP_DLL_TARGET})
+  if (WIN32)
+    list(APPEND CSC_OPTIONS /define:CLR4)
+  endif ()
+  set(_PACKAGE_NAME ${_CSHARP_PACKAGE_NAME})
+
+  # Setup dependencies
+  set(_dlls)
+  set(_deps csharp_interface)
+  if (_CSHARP_DEPENDS)
+    foreach (_dep IN LISTS _CSHARP_DEPENDS)
+      file(TO_NATIVE_PATH "${CMAKE_CURRENT_BINARY_DIR}/${_dep}" _dep)
+      get_filename_component(_filename ${_dep} NAME)
+      list(APPEND _dlls ${_filename}.dll)
+      list(APPEND CSC_OPTIONS /r:${_dep}.dll)
+    endforeach ()
+  endif (_CSHARP_DEPENDS)
+  if (_CSHARP_SYSTEM_DEPENDS)
+    foreach (_dep IN LISTS _CSHARP_SYSTEM_DEPENDS)
+      if (_dep MATCHES "^OSGeo")
+        list(APPEND _deps ${_dep})
+      endif ()
+    endforeach ()
+  endif ()
+
+  # run the c# build
+  if (DOTNET_FOUND)
+    configure_file("${CMAKE_CURRENT_SOURCE_DIR}/dll_template.csproj"
+                   "${CMAKE_CURRENT_BINARY_DIR}/${_CSHARP_TARGET_SUBDIR}/${_root}.csproj.in")
+    file(
+      GENERATE
+      OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${_CSHARP_TARGET_SUBDIR}/${_root}-$<CONFIG>.csproj
+      INPUT ${CMAKE_CURRENT_BINARY_DIR}/${_CSHARP_TARGET_SUBDIR}/${_root}.csproj.in)
+    add_custom_command(
+      COMMAND ${CMAKE_COMMAND} -E "copy_if_different"
+              "${CMAKE_CURRENT_BINARY_DIR}/${_CSHARP_TARGET_SUBDIR}/${_root}-$<CONFIG>.csproj" "${_CSHARP_PROJ}"
+      VERBATIM PRE_BUILD
+      DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/${_CSHARP_TARGET_SUBDIR}/${_root}-$<CONFIG>.csproj"
+      OUTPUT ${_CSHARP_PROJ})
+    add_dotnet(
+      ${_CSHARP_PROJ}
+      ${CSHARP_CONFIG}
+      NAME
+      ${_PACKAGE_NAME}
+      VERSION
+      ${_VERSION_STRING}
+      OUTPUT_PATH
+      ${_CSHARP_PROJ_PATH}
+      DEPENDS
+      ${_deps}
+      PACKAGE
+      ${_CSHARP_SYSTEM_DEPENDS})
+
+  else ()
+    message("BUILDING : " ${_CSHARP_TARGET} " : " ${CSHARP_COMPILER} ${CSC_OPTIONS} /out:${_CSHARP_TARGET})
```

Comment:
could be removed or commented to avoid too verbose log

Reply:
Made into --verbose only